### PR TITLE
AlexCode ability to schedule the post in the future + draft&future posts

### PR DESCRIPTION
**Pull request by ntorbinskiy**
AlexCode ability to schedule the post in the future

### CODE REVIEW

**Comment by 1alexvash** [comment]
@ntorbinskiy 
You call the getAllPostDocuments function twice, just apply filtering in the method definition to not repeat filtering logic

**Comment by 1alexvash** [comment]
@ntorbinskiy and refactor the filtering inside a separate method, as was recommended in the Todoist task, you are going to need this method for another upcoming task

**Comment by ntorbinskiy** [comment]
@1alexvash done. Waiting of code review

File: helpers/markdownDocumentsReader.ts [hunk 1]
```diff
@@ -18,16 +18,22 @@ export function getPostDocumentBySlug(slug: string) {
...
-    .map((post) => {
-      const { content, ...postWithoutContent } = post;
-      return postWithoutContent;
-    });
+    .sort((post1, post2) => (post1.date > post2.date ? -1 : 1));
+
+  const filteredPostsByDate = isPostInTheFuture(docs).map((post) => {
+    const { content, ...postWithoutContent } = post;
+    return postWithoutContent;
+  });
```
[the remark this answers is not in the document]
Comment (ntorbinskiy) [reply helpers/markdownDocumentsReader.ts:36]: @1alexvash  what do you meen "second filter in the row"?

[the remark this answers is not in the document]
Comment (1alexvash) [reply helpers/markdownDocumentsReader.ts:36]: @ntorbinskiy assigning second variable in the row for filtering

image

File: helpers/markdownDocumentsReader.ts [hunk 2]
```diff
@@ -18,16 +18,22 @@ export function getPostDocumentBySlug(slug: string) {
   return JSONSerialize({ slug: realSlug, ...data, content }) as PostDocument;
 }
 
+const today = new Date();
+
+const isPostInTheFuture = (posts: PostDocument[]) =>
+  posts.filter((item) => today.toISOString() > item.date);
```
Comment (1alexvash) [inline helpers/markdownDocumentsReader.ts:24]:
```suggestion
const isPostInTheFuture = (post: PostDocument) =>
  if ((post.date) => today.toISOString() > item.date) {
    return true;
  } else {
    return false;
  }
```

@ntorbinskiy is it clear now?

File: components/PostCard.tsx [hunk 3]
```diff
@@ -8,6 +8,36 @@ interface Props {
...
+};
+
+const FuturePost = () => {
+  return (
+    <div className="future-post">
+      <div className="triangle-future triangle triangle-item">
+        <span className="triangle-text-future triangle-text">FUTURE</span>
+        <span className="triangle-text-future-post triangle-text"> POST</span>
+      </div>
+      <div className="small-triagles-future">
```
Comment (1alexvash) [inline components/PostCard.tsx:32]: @ntorbinskiy check spelling

File: components/PostCard.tsx [hunk 4]
```diff
@@ -8,6 +8,36 @@ interface Props {
...
+        <div className="small-triagles-item-future1 triangle small-triagles-item-future small-triagles-item"></div>
+        <div className="small-triagles-item-future2 triangle small-triagles-item-future small-triagles-item"></div>
```
Comment (1alexvash) [inline components/PostCard.tsx:34]: @ntorbinskiy discussed this thing in the morning, find a better name for 1, and 2

File: components/PostCard.tsx [hunk 5]
```diff
@@ -8,6 +8,36 @@ interface Props {
   post: PostDocumentWithoutContent;
 }
 
+const DraftPost = () => {
+  return (
```
Comment (1alexvash) [inline components/PostCard.tsx:12]: @ntorbinskiy Use implicit return

File: components/PostCard.tsx [hunk 6]
```diff
@@ -8,6 +8,36 @@ interface Props {
...
+      <div className="small-triagles-draft">
+        <div className="small-triagles-item-draft1 triangle small-triagles-item-draft small-triagles-item"></div>
+        <div className="small-triagles-item-draft2 triangle small-triagles-item-draft small-triagles-item"></div>
+      </div>
+    </div>
+  );
+};
+
+const FuturePost = () => {
+  return (
```
Comment (1alexvash) [inline components/PostCard.tsx:26]: Same story

File: components/PostCard.tsx [hunk 7]
```diff
@@ -28,9 +58,16 @@ const toBase64 = (str: string) => Buffer.from(str).toString("base64");
 const PostCard = ({ post }: Props) => {
   const dispatch = useAppDispatch();
 
+  const isFuturePost = today.toISOString() > post.date ? false : true;
```
Comment (1alexvash) [inline components/PostCard.tsx:61]: What's that?

Why do you duplicate helper function?

`helpers/checkOfDraftOrFuturePost.ts`

File: components/PostCard.tsx [hunk 8]
```diff
@@ -28,9 +58,16 @@ const toBase64 = (str: string) => Buffer.from(str).toString("base64");
...
+  const classNameOfPost =
+    post.draft === true || isFuturePost === true
+      ? "posts-list-block posts-list-block-draft-or-future"
+      : "posts-list-block";
   return (
     <li>
-      <div className="posts-list-block">
+      {post.draft ? <DraftPost /> : null}
```
Comment (1alexvash) [inline components/PostCard.tsx:68]: Same story

File: helpers/markdownDocumentsReader.ts [hunk 9]
```diff
@@ -17,12 +19,21 @@ export function getPostDocumentBySlug(slug: string) {
...
-
 export function getAllPostDocuments(): PostDocumentWithoutContent[] {
   const slugs = fs.readdirSync(documentsDirectory);
   const docs = slugs
     .map((slug) => getPostDocumentBySlug(slug))
-    .filter((post: PostDocument) => post.draft === false)
+    .filter((post: PostDocument) => {
+      {
+        if (process.env.NODE_ENV === "development") {
+          return post;
```
Comment (1alexvash) [inline helpers/markdownDocumentsReader.ts:29]:
```suggestion
          return true;
```
It's a `filter` not `map`

File: styles/main.scss [hunk 10]
```diff
@@ -1408,3 +1403,99 @@ blockquote p:last-child {
...
+
+.triangle-text {
+  display: block;
+  position: relative;
+  transform: rotate(88deg);
+  font-weight: 500;
+  color: white;
+}
+
+.small-triagles-item {
```
Comment (1alexvash) [inline styles/main.scss:1427]: @ntorbinskiy 
spelling

File: components/PostCard.tsx [hunk 11]
```diff
@@ -8,6 +8,36 @@ interface Props {
...
+      </div>
+      <div className="small-triagles-future">
+        <div className="small-triagles-item-future1 triangle small-triagles-item-future small-triagles-item"></div>
+        <div className="small-triagles-item-future2 triangle small-triagles-item-future small-triagles-item"></div>
+      </div>
+    </div>
+  );
+};
+
+const today = new Date();
```
Comment (1alexvash) [inline components/PostCard.tsx:40]:
```suggestion
const today = new Date();

```

File: components/PostCard.tsx [hunk 12]
```diff
@@ -28,9 +57,16 @@ const toBase64 = (str: string) => Buffer.from(str).toString("base64");
...
+  const classNameOfPost =
+    isPostADraft(post) === true || isPostInTheFuture(post) === false
+      ? "posts-list-block posts-list-block-draft-or-future"
+      : "posts-list-block";
+
   return (
     <li>
-      <div className="posts-list-block">
+      {isPostADraft(post) ? <DraftPost /> : null}
+      {isPostInTheFuture(post) === false ? <FuturePost /> : null}
```
Comment (1alexvash) [inline components/PostCard.tsx:68]: @ntorbinskiy looks like && operator is going to work here

File: helpers/checkOfDraftOrFuturePost.ts [hunk 13]
```diff
@@ -0,0 +1,19 @@
+import { PostDocumentWithoutContent } from "interfaces";
+
+const today = new Date();
+
+export function isPostADraft(post: PostDocumentWithoutContent) {
+  if (post.draft === false) {
```
Comment (1alexvash) [inline helpers/checkOfDraftOrFuturePost.ts:6]: Start with the positive case first

File: helpers/markdownDocumentsReader.ts [hunk 14]
```diff
@@ -17,12 +19,21 @@ export function getPostDocumentBySlug(slug: string) {
...
 }
-
 export function getAllPostDocuments(): PostDocumentWithoutContent[] {
   const slugs = fs.readdirSync(documentsDirectory);
   const docs = slugs
     .map((slug) => getPostDocumentBySlug(slug))
-    .filter((post: PostDocument) => post.draft === false)
+    .filter((post: PostDocument) => {
+      {
+        if (process.env.NODE_ENV === "development") {
```
Comment (1alexvash) [inline helpers/markdownDocumentsReader.ts:28]: Add a const on the top of the function

const isDev = process.env.NODE_ENV === "development"

And then check for isDev only

File: styles/main.scss (hunk 10 above)
Comment (1alexvash) [reply styles/main.scss:1427]: I still seeing triagles

File: styles/main.scss [hunk 15]
```diff
@@ -1408,3 +1403,99 @@ blockquote p:last-child {
...
+  border-left: 5px solid rgb(0, 0, 0);
+  transform: rotate(45deg);
+}
+
+.small-triagles-item-draft1 {
+  top: -5px;
+  left: 80px;
+}
+
+.small-triagles-item-draft2 {
```
Comment (1alexvash) [inline styles/main.scss:1456]: here as well

File: components/Post/LatestPosts.tsx [hunk 16]
```diff
@@ -18,12 +18,20 @@ const LatestPosts = ({ latestPosts }: Props) => {
...
+                  layout="fill"
+                  objectFit="cover"
+                />
+              ) : (
+                <div className="draft-img" style={{ height: "102px" }}>
+                  <h1 className="draft-img-text" style={{ top: "35%" }}>
+                    draft
+                  </h1>
+                </div>
+              )}
```
Comment (1alexvash) [inline components/Post/LatestPosts.tsx:34]: @ntorbinskiy 

Looks cool, but I think we just can use this picture:

https://www.figma.com/file/d9V8ibM40YakvM3VFQDhfq/alex-code.com?node-id=1437%3A776

File: styles/main.scss [hunk 17]
```diff
@@ -1491,11 +1492,32 @@ blockquote p:last-child {
...
+
+.draft-img-text {
+  position: relative;
+  top: 50%;
+  color: black !important;
+  text-transform: uppercase;
+  font-weight: 500;
+  font-size: 25px;
+  text-align: center;
+}
```
Comment (1alexvash) [inline styles/main.scss:1519]: Probably, you're not going to need this class, if you just pass a Figma image

File: components/PostCard.tsx [hunk 18]
```diff
@@ -28,22 +58,36 @@ const toBase64 = (str: string) => Buffer.from(str).toString("base64");
...
+  const classNameOfPost =
+    isPostADraft(post) === true || isPostInTheFuture(post) === false
+      ? "posts-list-block posts-list-block-draft-or-future"
+      : "posts-list-block";
+
   return (
     <li>
-      <div className="posts-list-block">
+      {isPostADraft(post) ? <DraftPostMark /> : null}
+      {isPostInTheFuture(post) === false ? <FuturePostMark /> : null}
```
Comment (1alexvash) [inline components/PostCard.tsx:69]: It looks like my comment, is lost, but why not && ?

File: components/PostCard.tsx [hunk 19]
```diff
@@ -28,22 +58,36 @@ const toBase64 = (str: string) => Buffer.from(str).toString("base64");
...
-                layout="fill"
-                objectFit="cover"
-                placeholder="blur"
-                blurDataURL={`data:image/svg+xml;base64,${toBase64(
-                  shimmer(378, 378)
-                )}`}
-              />
+              {post.featuredImage ? (
+                <Image
+                  src={post.featuredImage}
```
Comment (1alexvash) [inline components/PostCard.tsx:76]: If no featured image, pass image placeholder from Figma

File: components/PostCard.tsx [hunk 20]
```diff
@@ -28,22 +58,36 @@ const toBase64 = (str: string) => Buffer.from(str).toString("base64");
...
+                  alt="blog post image"
+                  layout="fill"
+                  objectFit="cover"
+                  placeholder="blur"
+                  blurDataURL={`data:image/svg+xml;base64,${toBase64(
+                    shimmer(378, 378)
+                  )}`}
+                  style={{ filter: "grayscale(50%)" }}
```
Comment (1alexvash) [inline components/PostCard.tsx:84]: image
Why every picture is greyed out right now?

It should be only for upcoming posts

File: styles/main.scss [hunk 21]
```diff
@@ -1408,3 +1403,121 @@ blockquote p:last-child {
...
+
+.gray-filter-for-img {
+  filter: grayscale(50%);
```
Comment (1alexvash) [inline styles/main.scss:1522]: image
Why you don't use this class here?

It creates code duplication

File: components/PostCard.tsx (hunk 18 above)
Comment (ntorbinskiy) [reply components/PostCard.tsx:69]: bcs post can be in the future but no draft. Or don't u mean that? 
```suggestion
      {isPostADraft(post) && isPostInTheFuture(post) === false ? <DraftPostMark /> <FuturePostMark /> : null}
```

Comment (1alexvash) [reply components/PostCard.tsx:69]: Not that, it just a comment got lost

I was suggesting the following change:

```ts
{isPostADraft(post) && <DraftPostMark />}
{isPostInTheFuture(post) === false && <FuturePostMark />}
```

So we don't have write a ternary case for null value

If there are any problems with this method, it's cool to skip it

Comment (1alexvash) [reply components/PostCard.tsx:69]: @ntorbinskiy

File: components/PostCard.tsx [hunk 22]
```diff
@@ -28,21 +58,37 @@ const toBase64 = (str: string) => Buffer.from(str).toString("base64");
...
                 objectFit="cover"
                 placeholder="blur"
                 blurDataURL={`data:image/svg+xml;base64,${toBase64(
                   shimmer(378, 378)
                 )}`}
+                style={
+                  classNameOfPost.includes("posts-list-block-draft-or-future")
+                    ? { filter: "grayscale(50%)" }
+                    : {}
+                }
```
Comment (1alexvash) [inline components/PostCard.tsx:91]: @ntorbinskiy 

There's a better way to check it:

`isPostADraft(post) || isPostInTheFuture`

File: content/posts/2022-10-08-why-good-code-style-matters.md [hunk 23]
```diff
@@ -1,8 +1,10 @@
 ---
 title: Why good code style matters
 date: 2022-10-24T10:10:19.885Z
-featuredImage: /post-images/good-style-article-prettier.png
-draft: false
+
+featuredImage:
```
Comment (1alexvash) [inline content/posts/2022-10-08-why-good-code-style-matters.md:5]: @ntorbinskiy It is only for local testing? It should not be in the PR to break things

File: helpers/checkOfDraftOrFuturePost.ts (hunk 13 above)
Comment (1alexvash) [reply helpers/checkOfDraftOrFuturePost.ts:6]: @ntorbinskiy

File: helpers/markdownDocumentsReader.ts (hunk 14 above)
Comment (1alexvash) [reply helpers/markdownDocumentsReader.ts:28]: @ntorbinskiy

File: components/Post/PostContent.tsx [hunk 24]
```diff
@@ -31,9 +37,14 @@ const PostContent = ({ post }: Props) => {
       </Head>
 
       <div className="blogpost-image">
+        {isPostADraft(post) ? <DraftPostMark /> : null}
+        {isPostInTheFuture(post) === false ? <FuturePostMark /> : null}
```
Comment (1alexvash) [inline components/Post/PostContent.tsx:41]: @ntorbinskiy 🤯

If post NOT in the future, then show FuturePostMark

It makes no sense

Comment (1alexvash) [reply components/Post/PostContent.tsx:41]: And is there a specific reason you use ternary instead of && operator?

File: components/Post/PostContent.tsx [hunk 25]
```diff
@@ -31,9 +37,14 @@ const PostContent = ({ post }: Props) => {
...
         <img
-          src={post.featuredImage ?? "/post-images/placeholder.png"}
+          src={
+            post.featuredImage ? post.featuredImage : "/post-images/draft.webp"
+          }
           alt="blog post image"
+          className="gray-filter-for-img"
```
Comment (1alexvash) [inline components/Post/PostContent.tsx:47]: image
This pictures, has this class, but does not make any effect 🤯

File: components/PostCard.tsx [hunk 26]
```diff
@@ -1,13 +1,43 @@
...
 }
 
+export const DraftPostMark = () => (
+  <div className="draft-post">
+    <div className="triangle-draft triangle triangle-item">
+      <span className="triangle-text-draft triangle-text">draft</span>
+    </div>
+    <div className="small-triangles-draft">
+      <div className="small-triangles-item-draft-first triangle small-triangles-item-draft small-triangles-item"></div>
+      <div className="small-triangles-item-draft-second triangle small-triangles-item-draft small-triangles-item"></div>
```
Comment (1alexvash) [inline components/PostCard.tsx:23]: image
Make the name of this class at least 2 or 3 times shorter,
90 characters is overkill

File: components/Post/PostContent.tsx [hunk 27]
```diff
@@ -31,11 +36,22 @@ const PostContent = ({ post }: Props) => {
...
+          src={
+            post.featuredImage ? post.featuredImage : "/post-images/draft.webp"
+          }
           alt="blog post image"
           width={790}
           height={394}
+          style={
+            isPostADraft(post) || isPostInTheFuture(post)
+              ? { filter: "grayscale(50%)", borderRadius: "3px" }
+              : //Added borderRadius, because it haven't been working
```
Comment (alexander-vashchuk) [inline components/Post/PostContent.tsx:52]: Remove comment

File: components/PostCard.tsx [hunk 28]
```diff
@@ -1,13 +1,43 @@
...
 
+export const DraftPostMark = () => (
+  <div className="drt-post">
+    <div className="trn-drt trn trn-i">
+      <span className="trn-txt-drt trn-txt">draft</span>
+    </div>
+    <div className="sm-trns-drt">
+      <div className="sm-trns-i-drt-first trn sm-trns-i-drt sm-trns-i"></div>
+      <div className="sm-trns-i-drt-sec trn sm-trns-i-drt sm-trns-i"></div>
+    </div>
```
Comment (alexander-vashchuk) [inline components/PostCard.tsx:24]: WTF is drt?
Has this code been compiled? If so, it is not supposed to be read by a human

Otherwise, give it a human readable name

File: components/PostCard.tsx [hunk 29]
```diff
@@ -1,13 +1,43 @@
...
+    <div className="trn-fut trn trn-i">
+      <span className="trn-txt-fut trn-txt">future</span>
+      <span className="trn-txt-fut-post trn-txt"> post</span>
+    </div>
+    <div className="sm-trns-fut">
+      <div className="sm-trns-i-fut-first trn sm-trns-i-fut sm-trns-i"></div>
+      <div className="sm-trns-i-fut-sec trn sm-trns-i-fut sm-trns-i"></div>
+    </div>
+  </div>
+);
```
Comment (alexander-vashchuk) [inline components/PostCard.tsx:39]: Same story, just git revert commit with renaming

File: helpers/checkOfDraftOrFuturePost.ts [hunk 30]
```diff
@@ -0,0 +1,19 @@
...
+  }
+}
+
+export function isPostInTheFuture(post: PostDocumentWithoutContent) {
+  if (today.toISOString() < post.date) {
+    return true;
+  } else {
+    return false;
+  }
+}
```
Comment (alexander-vashchuk) [inline helpers/checkOfDraftOrFuturePost.ts:19]: Still not quite right:

image

Maybe because toISOString converts type, and it fucks the comparison

File: helpers/markdownDocumentsReader.ts [hunk 31]
```diff
@@ -22,7 +26,17 @@ export function getAllPostDocuments(): PostDocumentWithoutContent[] {
...
   const docs = slugs
     .map((slug) => getPostDocumentBySlug(slug))
-    .filter((post: PostDocument) => post.draft === false)
+    .filter((post: PostDocument) => {
+      {
+        if (isDev) {
+          return true;
+        } else {
+          return (
+            isPostADraft(post) === false && isPostInTheFuture(post) === false
```
Comment (alexander-vashchuk) [inline helpers/markdownDocumentsReader.ts:35]: In return, you return another condition to check 🤯

Comment (alexander-vashchuk) [reply helpers/markdownDocumentsReader.ts:35]: Create another if condition for it

File: styles/main.scss [hunk 32]
```diff
@@ -640,7 +635,7 @@ header {
   left: 0;
   bottom: 0;
   width: 100%;
-  height: 66%;
+  height: 50%;
```
Comment (alexander-vashchuk) [inline styles/main.scss:638]: Why this line was changed?

File: styles/main.scss [hunk 33]
```diff
@@ -623,6 +614,10 @@ header {
   pointer-events: none;
 }
 
+.posts-list-block:hover .content:after {
+  border-color: #fe6c0a;
```
Comment (alexander-vashchuk) [inline styles/main.scss:618]: What this line does?

File: styles/main.scss [hunk 34]
```diff
@@ -604,10 +599,6 @@ header {
   border: 1px solid transparent;
 }
 
-.posts-list-block:hover .content:after {
-  border-color: #fe6c0a;
-}
-
```
Comment (alexander-vashchuk) [inline styles/main.scss:610]: Why were those lines removed?

File: components/Post/PostContent.tsx [hunk 35]
```diff
@@ -1,8 +1,13 @@
...
+import {
+  isPostADraft,
+  isPostInTheFuture,
+} from "helpers/checkOfDraftOrFuturePost";
 import toHumanReadableDate from "helpers/toHumanReadableDate";
 import { PostDocument } from "interfaces";
 import Head from "next/head";
 
+import { DraftPostMark, FuturePostMark } from "../PostCard";
 interface Props {
```
[the remark this answers is not in the document]
Comment (ntorbinskiy) [reply components/Post/PostContent.tsx:11]: No, I just haven't setup ESLINT correctly yet

File: components/PostCard.tsx (hunk 28 above)
Comment (ntorbinskiy) [reply components/PostCard.tsx:24]: I have written about this problem in our telegram chat. I don't understand, how I am supposed to change the name to the 
shorter in 2 or 3 times and save the readability of it.  I can delete all this classes, but then, we will have a problem of duplicates of styles

File: styles/main.scss (hunk 32 above)
Comment (ntorbinskiy) [reply styles/main.scss:638]: Sorry, mistake

File: styles/main.scss (hunk 34 above)
Comment (ntorbinskiy) [reply styles/main.scss:610]: That snippet has just been removed a little bit lower, for better readability, you mentioned it at comment above🤔

File: components/PostCard.tsx [hunk 36]
```diff
@@ -28,21 +58,38 @@ const toBase64 = (str: string) => Buffer.from(str).toString("base64");
...
                 objectFit="cover"
                 placeholder="blur"
                 blurDataURL={`data:image/svg+xml;base64,${toBase64(
                   shimmer(378, 378)
                 )}`}
+                style={
+                  isPostADraft(post) || isPostInTheFuture(post)
+                    ? { filter: "grayscale(50%)" }
+                    : {}
+                }
```
[the remark this answers is not in the document]
Comment (ntorbinskiy) [reply components/PostCard.tsx:92]: image
Have I rigthly understood you?
What is "condotion" at your example of code?

File: components/PostCard.tsx (hunk 29 above)
Comment (ntorbinskiy) [reply components/PostCard.tsx:39]: Wrote above about it

File: styles/main.scss (hunk 32 above)
Comment (ntorbinskiy) [reply styles/main.scss:638]: @1alexvash

File: styles/main.scss (hunk 34 above)
Comment (ntorbinskiy) [reply styles/main.scss:610]: @1alexvash

File: components/PostCard.tsx (hunk 36 above)
[the remark this answers is not in the document]
Comment (ntorbinskiy) [reply components/PostCard.tsx:92]: @1alexvash

File: components/PostCard.tsx (hunk 29 above)
Comment (ntorbinskiy) [reply components/PostCard.tsx:39]: @1alexvash

File: components/PostCard.tsx (hunk 28 above)
Comment (ntorbinskiy) [reply components/PostCard.tsx:24]: @1alexvash

File: components/Post/PostContent.tsx (hunk 35 above)
[the remark this answers is not in the document]
Comment (ntorbinskiy) [reply components/Post/PostContent.tsx:11]: @1alexvash

File: components/Post/LatestPosts.tsx [hunk 37]
```diff
@@ -19,10 +23,19 @@ const LatestPosts = ({ latestPosts }: Props) => {
...
+                    : "/post-images/draft.webp"
+                }
                 alt="blog post image"
                 layout="fill"
                 objectFit="cover"
+                style={
+                  isPostADraft(post) || isPostInTheFuture(post)
+                    ? { filter: "grayscale(50%)" }
+                    : {}
+                }
```
[the remark this answers is not in the document]
Comment (ntorbinskiy) [reply components/Post/LatestPosts.tsx:38]: Wrote lower about it

[the remark this answers is not in the document]
Comment (ntorbinskiy) [reply components/Post/LatestPosts.tsx:38]: @1alexvash

File: components/Post/PostContent.tsx (hunk 35 above)
[the remark this answers is not in the document]
Comment (1alexvash) [reply components/Post/PostContent.tsx:11]: We don't have ESLint rule for having break lines after import statements,
It just you being clumsy 😐

File: components/Post/PostContent.tsx (hunk 27 above)
Comment (1alexvash) [reply components/Post/PostContent.tsx:52]: What this like mean?

You like some of those comments, and then just skip some of the comments

File: components/PostCard.tsx (hunk 28 above)
Comment (1alexvash) [reply components/PostCard.tsx:24]: @ntorbinskiy In telegram I answered to return previous classes, and in this comment I said to use "git revert" command for it, what's not clear on that one?

File: components/PostCard.tsx (hunk 29 above)
Comment (1alexvash) [reply components/PostCard.tsx:39]: answered

File: components/PostCard.tsx (hunk 36 above)
[the remark this answers is not in the document]
Comment (1alexvash) [reply components/PostCard.tsx:92]: It's correct

File: content/posts/2022-10-08-why-good-code-style-matters.md [hunk 38]
```diff
@@ -1,8 +1,10 @@
...
-date: 2022-10-24T10:10:19.885Z
+date: 2023-10-24T10:10:19.885Z
+
 featuredImage: /post-images/good-style-article-prettier.png
-draft: false
+
+draft: true
 tags:
   - CleanCode
 ---
```
Comment (1alexvash) [inline content/posts/2022-10-08-why-good-code-style-matters.md:10]: Leave some comment around it, like "This code is for testing only DO NOT MERGE IT"
So it doesn't get accidentaly merged

File: styles/main.scss (hunk 34 above)
Comment (1alexvash) [reply styles/main.scss:610]: Which snippet
?

File: components/Post/LatestPosts.tsx (hunk 37 above)
[the remark this answers is not in the document]
Comment (1alexvash) [reply components/Post/LatestPosts.tsx:38]: Answered, do the same thing

File: components/Post/PostContent.tsx (hunk 27 above)
Comment (ntorbinskiy) [reply components/Post/PostContent.tsx:52]: I like it for myself. It's like checklist, so I'll know, that I answered to the each of your comments

File: components/Post/PostContent.tsx (hunk 35 above)
[the remark this answers is not in the document]
Comment (ntorbinskiy) [reply components/Post/PostContent.tsx:11]: Oh, okay😄

File: styles/main.scss (hunk 32 above)
[the remark this answers is not in the document]
Comment (ntorbinskiy) [reply styles/main.scss:638]: @1alexvash  wrote in telegram about it, check, please

File: styles/main.scss (hunk 34 above)
Comment (ntorbinskiy) [reply styles/main.scss:610]: .posts-list-block:hover .content:after {
  border-color: #fe6c0a;
}

File: components/PostCard.tsx (hunk 26 above)
Comment (1alexvash) [inline components/PostCard.tsx:23]: Remove those two divs, and just use CSS pseudo-elements
They are designed for such "decorated" things

Something like:
https://codepen.io/chriscoyier/pen/OmORqR

File: helpers/checkOfDraftOrFuturePost.ts [hunk 39]
```diff
@@ -0,0 +1,19 @@
...
+
+export function isPostInTheFuture(post: { date: Date | string }) {
+  if (today.toISOString() < post.date.toString()) {
+    return true;
+  } else {
+    return false;
+  }
+}
```
Comment (1alexvash) [inline helpers/checkOfDraftOrFuturePost.ts:19]: @ntorbinskiy did you test this function?

image

File: helpers/markdownDocumentsReader.ts [hunk 40]
```diff
@@ -22,7 +26,18 @@ export function getAllPostDocuments(): PostDocumentWithoutContent[] {
...
+      if (isDev) {
+        return true;
+      } else if (
+        isPostADraft(post) === false &&
+        isPostInTheFuture(post) === false
+      ) {
+        return true;
+      } else {
+        return false;
+      }
```
Comment (1alexvash) [inline helpers/markdownDocumentsReader.ts:39]:
```suggestion
      if (isDev) {
        return true;
      } else if (
        isPostADraft(post) === false &&
        isPostInTheFuture(post) === false
      ) {
        return true;
      } else {
        return false;
      }
```
if (isDev) {
        return true;
      }

      if (isPostADraft(post) || isPostInTheFuture(post)) {
        return false;
      }

      return true;

File: helpers/markdownDocumentsReader.ts [hunk 41]
```diff
@@ -22,7 +26,18 @@ export function getAllPostDocuments(): PostDocumentWithoutContent[] {
...
+    })
```
Comment (1alexvash) [inline helpers/markdownDocumentsReader.ts:40]:
```suggestion
    .filter((post: PostDocument) => {
      if (isDev) {
        return true;
      }

      if (isPostADraft(post) || isPostInTheFuture(post)) {
        return false;
      }

      return true;
    })
```

File: styles/main.scss [hunk 42]
```diff
@@ -579,10 +578,6 @@ header {
   overflow: hidden;
 }
 
-.posts-list-block:hover .post-img:after {
-  height: 100%;
-}
-
```
Comment (1alexvash) [inline styles/main.scss:585]: animation
hover effect is broken

File: styles/main.scss (hunk 34 above)
Comment (1alexvash) [reply styles/main.scss:610]: You just moved this piece of code?

File: helpers/checkOfDraftOrFuturePost.ts (hunk 39 above)
Comment (ntorbinskiy) [reply helpers/checkOfDraftOrFuturePost.ts:19]: Function works properly. When you give yesterday-date to the function - you don't translate it to ISO string. That's why, it isn't working
image
image

Comment (1alexvash) [reply helpers/checkOfDraftOrFuturePost.ts:24]: @ntorbinskiy 
So, you're comparing a string value with a number value, and you're saying it works correctly
?
image

Comment (1alexvash) [reply helpers/checkOfDraftOrFuturePost.ts:24]: image

Comment (ntorbinskiy) [reply helpers/checkOfDraftOrFuturePost.ts:24]: This works because we use Date syntaxis. For example, "2017/10/26" > "2017/10/7" gives false because we use "/" instead of "-"

Comment (ntorbinskiy) [reply helpers/checkOfDraftOrFuturePost.ts:24]: image
https://stackoverflow.com/questions/14781153/how-to-compare-two-string-dates-in-javascript

File: components/PostCard.tsx [hunk 43]
```diff
@@ -1,13 +1,35 @@
...
 
 import { setTags } from "../redux/slices/selectedTags";
+
 interface Props {
   post: PostDocumentWithoutContent;
 }
 
+export const DraftPostMark = () => (
+  <div className="draft-post">
+    <div className="triangle-draft triangle triangle-item">
```
Comment (1alexvash) [inline components/PostCard.tsx:18]: @ntorbinskiy 

Start class, with the main class triangle, and then add modificators

```suggestion
    <div className="triangle triangle-draft  triangle-item">
```

File: components/PostCard.tsx [hunk 44]
```diff
@@ -1,13 +1,35 @@
...
+      <span className="triangle-text-draft triangle-text">draft</span>
```
Comment (1alexvash) [inline components/PostCard.tsx:19]: Change the class name from:

triangle-text-draft => triangle-draft-text

File: content/posts/2022-10-08-why-good-code-style-matters.md [hunk 45]
```diff
@@ -1,8 +1,11 @@
...
 title: Why good code style matters
-date: 2022-10-24T10:10:19.885Z
+date: 2023-10-24T10:10:19.885Z
+
 featuredImage: /post-images/good-style-article-prettier.png
-draft: false
+
+draft: true
+# !This code is for testing only DO NOT MERGE IT!
```
Comment (1alexvash) [inline content/posts/2022-10-08-why-good-code-style-matters.md:8]: Cancel these chances, and then you can merge this branch by yourself

You have my approval
